Esri/arcgis-maps-sdk-swift-samples
# [New] Monitor changes to layer view state

## Description

This PR implements `Monitor changes to layer view state` in `Maps` category.
URL to README: [Monitor changes to layer view state](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-MonitorChangesToLayerViewState/Shared/Samples/Monitor%20changes%20to%20layer%20view%20state)

## Linked Issue(s)

- `swift/issues/5262`

## How To Test

- Interact with the sample and watch the status at the top of the screen change.
    - Toggle the layer to see "Active" and "Not Visible". 
    - Scale the map outside of the layer's scale range to see "Out of Scale".
    - Turn of your mac's network to see "Warning".

## Screenshots

![monitor-changes-to-layer-view-state](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/35ff3a0e-f911-4984-8486-530bb2aa7f0b)
